Documentation update for Lutron integration

cdheiser opened the pull request:
## Proposed change
Documentation update for https://github.com/home-assistant/core/pull/36902
- Lutron now supports config_flow
- And also now has an optional setting via the UI to use Areas defined by the Lutron RadioRA2 system.


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

cdheiser committed:
Update documentation for config_flow features in Lutrion

cdheiser committed:
Add documentation for the new lutron services

These services add support for pressing and releasing buttons which is useful as there are several features that don't map to entities that can be managed through buttons like the Alarm mode.

frenck commented on source/_integrations/lutron.markdown:
```diff
@@ -22,6 +23,8 @@ Presently, there's only support for communicating with the [RadioRA 2](http://ww
 
 When configured, the `lutron` integration will automatically discover the rooms and their associated switches/dimmers as configured by the RadioRA 2 software from Lutron. Each room will be treated as a separate group.
 
+This component supports importing Areas defined by the Lutron RadioRA2 system into Home Assistant.  You can enable this feature in the integration UI.
```
Please describe to the user how to do that. e.g., using click paths. Numerous examples can be found in our existing documentation.

cdheiser replied:
Done

frenck commented on source/_integrations/lutron.markdown:
```diff
@@ -22,6 +23,8 @@ Presently, there's only support for communicating with the [RadioRA 2](http://ww
 
+This component supports importing Areas defined by the Lutron RadioRA2 system into Home Assistant.  You can enable this feature in the integration UI.
+
 To use Lutron RadioRA 2 devices in your installation, add the following to your `configuration.yaml` file using the IP address of your RadioRA 2 main repeater:
```
Isn't this deprecated now?

cdheiser replied:
I was looking through other component documentation trying to figure out what the standard is for describing the configuration migration to config_flow.  If you have a good example I'm happy to model it after that.  This came as a result of looking at some of the ones that I know still support configuration.yaml and have config_flow like rainmachine and tplink.

frenck commented on source/_integrations/lutron.markdown:
```diff
@@ -22,6 +23,8 @@ Presently, there's only support for communicating with the [RadioRA 2](http://ww
 
 When configured, the `lutron` integration will automatically discover the rooms and their associated switches/dimmers as configured by the RadioRA 2 software from Lutron. Each room will be treated as a separate group.
 
+This component supports importing Areas defined by the Lutron RadioRA2 system into Home Assistant.  You can enable this feature in the integration UI.
```
```suggestion
# Configuration

This component supports importing Areas defined by the Lutron RadioRA2 system into Home Assistant.  You can enable this feature in the integration UI.
```

cdheiser replied:
I rewrote this section.  PTAL

frenck commented on source/_integrations/lutron.markdown:
```diff
@@ -53,6 +56,32 @@ It is recommended to assign a static IP address to your main repeater. This ensu
+
+### `lutron.press_button`
+
+Press a Keypad button.  This is useful for the Alarm virtual button where pressing the button activates alarm mode and releasing the button clears the alarm mode.
```
```suggestion
Press a Keypad button. This is useful for the Alarm virtual button were pressing the button activates alarm mode and releasing the button clears the alarm mode.
```

frenck reviewed (changes requested):
Thanks for the PR @cdheiser!
I've reviewed it and left some comments. Could you take a look? Thanks! 👍

cdheiser committed:
Update source/_integrations/lutron.markdown

Spelling fix

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

cdheiser committed:
Change documentation around configuration

Indicate that configuration.yaml is deprecated
Include directions on how to find the integration.

cdheiser commented:
@frenck do you have more thoughts or pointers for getting the documentation consistent?

frenck reviewed (approved):
✅ Approved. Can be merged as soon as the parent PR gets merged.

frenck commented:
Closing PR as the parent PR has been closed.
